### ISSUE TITLE
Preprocessing of docx template?

### ISSUE THREAD

**Issue by koosvanderkolk** [guigrpa/docx-templates#359]
My use case is that I would like to let 'less technical' end-users create their own templates.

E.g. let them create

![image](https://github.com/guigrpa/docx-templates/assets/662014/de5dcea6-f82a-4890-ad67-698e99fd4ebd)

Which should then be converted to:

![image](https://github.com/guigrpa/docx-templates/assets/662014/89e76554-e492-4259-8243-d5aa73b3efe1)

I also have quite some variables which contain HTML. For those, I want the end-user to just add 

![image](https://github.com/guigrpa/docx-templates/assets/662014/b08754d0-29ed-40b9-a092-3431669ebeee)

Which should then be converted to

![image](https://github.com/guigrpa/docx-templates/assets/662014/6e794c37-435e-4a6e-8d0b-80e3fbcd21a6)

So the feature request would be to have a preprocess function, in which I can alter the template before it is being interpreted.

**Comment by jjhbw** [comment]
I don't think it is wise to implement this in docx-templates, as that would add yet another layer of configuration and complexity.

You can probably do 90% of what you want by using a custom `runJs` function. See e.g. https://github.com/guigrpa/docx-templates/blob/5ad3cfd40866961a99a3772b07ad1cdea8b9b23c/examples/example-vm2/index.js#L40
